test/zipkin/tracer_test.cc

```diff
@@ -0,0 +1,168 @@
+#include "zipkin/tracer.h"
+#include "zipkin/util.h"
+#include "zipkin/zipkin_core_constants.h"
+
+#include "gtest/gtest.h"
+
+namespace Zipkin {
+
+TEST(ZipkinTracerTest, spanCreation) {
+  Tracer tracer("my_service_name", "127.0.0.1:9000");
+  int64_t timestamp = Util::timeSinceEpochMicro();
+
+  // ==============
+  // Test the creation of a root span --> CS
+  // ==============
+
+  Span root_span = tracer.startSpan("my_span", timestamp);
+
+  EXPECT_EQ("my_span", root_span.name());
+  EXPECT_EQ(timestamp, root_span.startTime());
+
+  EXPECT_NE(0ULL, root_span.traceId());           // trace id must be set
+  EXPECT_EQ(root_span.traceId(), root_span.id()); // span id and trace id must be the same
+  EXPECT_FALSE(root_span.isSet().parent_id);      // no parent set
+  EXPECT_NE(0LL, root_span.timestamp());          // span's timestamp must be set
+
+  // A CS annotation must have been added
+  EXPECT_EQ(1ULL, root_span.annotations().size());
+  Annotation ann = root_span.annotations()[0];
+  EXPECT_EQ(ZipkinCoreConstants::CLIENT_SEND, ann.value());
+  EXPECT_NE(0ULL, ann.timestamp()); // annotation's timestamp must be set
+  EXPECT_TRUE(ann.isSetEndpoint());
+  Endpoint endpoint = ann.endpoint();
+  EXPECT_EQ("127.0.0.1", endpoint.ipv4());
```

Comment:
Correct. Since the current implementation supports only JSON over HTTP, we are storing IP addresses as strings. When we support other transports, we will revisit that (and possibly a few other aspects of the core data structures).